core/services/head_tracker.go

```diff
@@ -451,6 +451,27 @@ func (ht *HeadTracker) subscribeToHead() error {
 	return nil
 }
 
+type htUnsubscriber interface {
+	Unsubscribe()
+}
+
+// timedUnsubscribe attempts to unsubscribe but aborts abruptly after a time delay
+// unblocking the application. This is an effort to mitigate the occasional
+// indefinite block described here from go-ethereum:
+// https://chainlink/pull/600#issuecomment-426320971
```

Comment:
This link does not work

Reply:
Looks like you didnt change it though, only moved it?

Reply:
Exactly. Tbh, I didn't even look closely at that link. I'll fix it